cornice_swagger/swagger.py

```diff
@@ -383,6 +393,86 @@ def _ref(self, param, depth, base_name=None):
         return {'$ref': pointer}
 
 
+class ResponseHandler(object):
+    """Handles swagger response definitions."""
+
+    json_pointer = '#/responses/'
+
+    def __init__(self, definition_handler=DefinitionHandler(), ref=False):
+        """
+        :param definition_handler:
+            Callable that handles swagger definition schemas.
+        :param ref:
+            Specifies the ref value when calling from_xxx methods.
+        """
+
+        self.responses = {}
+
+        self.definitions = definition_handler
+        self.ref = ref
+
+    def from_schema_mapping(self, schema_mapping):
+        """
+        Creates a Swagger response object from a dict of response schemas.
+
+        :param schema_mapping:
+            Dict with entries matching ``{status_code: response_schema}``.
+        :rtype: dict
+            Response schema.
+        """
+
+        responses = {}
+
+        for status, response_schema in schema_mapping.items():
+
+            response = {}
+            if response_schema.description:
+                response['description'] = response_schema.description
+            else:
+                raise CorniceSwaggerException('Responses must have a description.')
+
+            for field_schema in response_schema.children:
+                location = field_schema.name
+                if location == 'body':
+                    field_schema.title = field_schema.__class__.__name__
+                    response['schema'] = self.definitions.from_schema(field_schema)
+                elif location == 'headers':
+                    headers = convert_schema(field_schema)
+                    if 'properties' in headers:
+                        response['headers'] = headers['properties']
+
+            pointer = response_schema.__class__.__name__
+            response = self._ref(response, self.ref, pointer)
+            responses[status] = response
+
+        return responses
+
+    def _ref(self, resp, depth, base_name=None):
+        """
+        Map responses in the responses section using JSON pointers.
+
+        :param schema:
+            Swagger response definition.
+        :param depth:
+            If param should be set as a reference. True means it should.
+        :param base_name:
+            Name that should be used for the reference.
+
+        :rtype: dict
+            Swagger schema.
```

Comment:
Ah, I hadn't read/reread those classes. Well, I think `_ref` in `DefinitionHandler` is somewhat different from this `_ref`, so I don't think it's important to maintain compatibility. In fact, I might rename `_ref` in `DefinitionHandler` to something like `_ref_recursive` or `_extract_refs`.